tests/utils.ts

```diff
@@ -0,0 +1,74 @@
+import { expect, Page, test } from "@playwright/test";
+
+export const joinRoomAndAddScreenShare = async (page: Page, roomId: string): Promise<string> =>
+  test.step("Join room and add track", async () => {
+    const peerRequest = await createPeer(page, roomId);
+    try {
+      const {
+        peer: { id: peerId },
+        token: peerToken,
+      } = (await peerRequest.json()).data;
+
+      await test.step("Join room", async () => {
+        await page.getByPlaceholder("token").fill(peerToken);
+        await page.getByRole("button", { name: "Connect", exact: true }).click();
+        await expect(page.getByText("Status: joined")).toBeVisible();
+      });
+
+      await test.step("Add screenshare", async () => {
+        await page.getByRole("button", { name: "Start screen share", exact: true }).click();
+      });
+
+      return peerId;
+    } catch (e) {
+      // todo fix
+      throw { status: peerRequest.status(), response: await peerRequest.json() };
+    }
+  });
+
+export const assertThatRemoteTracksAreVisible = async (page: Page, otherClientIds: string[]) => {
+  await test.step("Assert that remote tracks are visible", () =>
+    Promise.all(
+      otherClientIds.map((peerId) => expect(page.locator(`css=video[data-peer-id="${peerId}"]`)).toBeVisible()),
+    ));
+};
+
+export const assertThatOtherVideoIsPlaying = async (page: Page) => {
+  await test.step("Assert that media is working", async () => {
+    const getDecodedFrames: () => Promise<number> = () =>
+      page.evaluate(async () => {
+        const peerConnection = (window as typeof window & { client: { webrtc: { connection: RTCPeerConnection } } })
+          .client.webrtc.connection;
+        const stats = await peerConnection.getStats();
+        for (const stat of stats.values()) {
+          if (stat.type === "inbound-rtp") {
+            return stat.framesDecoded;
+          }
+        }
+      });
+    const firstMeasure = await getDecodedFrames();
+    await expect(async () => (await getDecodedFrames()) > firstMeasure).toPass();
```

Comment:
I think `.toPass()` retries only if the callback throws an exception from inner `expect()`

https://playwright.dev/docs/test-assertions#expecttopass

i suggest something like:

```
await expect(async () => expect((await getDecodedFrames()) > firstMeasure).toBe(true)).toPass();
```